### PR TITLE
feat: add section reorder command

### PR DESCRIPTION
**Pull request by rmartins90**
## Overview

Adds `td section reorder` so sections can be reordered within a project from the CLI. It supports positional moves, plus relative moves before or after another section.

## Changes

- Added `td section reorder` with `--project`, `--section`, `--before`, `--after`, `--position`, `--json`, and `--dry-run`.
- Added a Sync API helper for Todoist's `section_reorder` command.
- Updated section tests, shared fixtures, `SKILL_CONTENT`, and the generated Todoist CLI skill file.

## Test plan

1. Create or pick a test project with at least three sections.
2. Run `node dist/index.js section reorder "Review" --project "<project>" --before "Done" --dry-run` and check the preview shows the expected order.
3. Run `node dist/index.js section reorder "Review" --project "<project>" --position 0 --json` and confirm the JSON output lists `Review` first.
4. List/open the project sections and confirm Todoist reflects the new order.

### CODE REVIEW

File: src/commands/section/reorder.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+async function loadProjectSections(api: TodoistApi, projectId: string): Promise<Section[]> {
+    const { results } = await paginate(
+        (cursor, limit) => api.getSections({ projectId, cursor: cursor ?? undefined, limit }),
+        { limit: Number.MAX_SAFE_INTEGER },
+    )
+    return results
+}
+
+function resolveSectionFromList(sections: Section[], ref: string): Section {
```
Comment (doistbot) [inline src/commands/section/reorder.ts:37]: [P2] This reinvents the in-memory entity resolution logic that already exists as `resolveFromList` in `src/lib/refs.ts`. Please export and reuse `resolveFromList` (updating it to handle exact-match ambiguity if that's an intentional improvement here) instead of duplicating the exact/partial matching and error-handling boilerplate.

File: src/commands/section/reorder.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+}
+
+export async function reorderSection(ref: string, options: ReorderSectionOptions): Promise<void> {
+    if (!options.project) {
+        throw new CliError('MISSING_PROJECT', 'Specify --project <ref> to reorder a section.', [
+            'Section names are scoped to a project.',
+        ])
+    }
+
+    const flagCount = [options.before, options.after, options.position].filter(
```
Comment (doistbot) [inline src/commands/section/reorder.ts]: [P3] The argument validation for mutually-exclusive placement flags is duplicated verbatim from `src/commands/project/reorder.ts`. Consider extracting a shared helper to validate reorder flags and avoid structural drift across commands.

File: src/commands/section/index.ts [hunk 3]
```diff
@@ -85,6 +87,40 @@ export function registerSectionCommand(program: Command): void {
...
+        .option('--json', 'Output the new ordering as JSON')
+        .option('--dry-run', 'Preview what would happen without executing')
+        .addHelpText(
+            'after',
+            `
+Examples:
+  td section reorder "Review" --project "Roadmap" --before "Done"
+  td section reorder "Review" --project "Roadmap" --after "In Progress"
+  td section reorder --section "Review" --project "Roadmap" --position 0
+  td section reorder "Review" --project "Roadmap" --position 0`,
```
Comment (doistbot) [inline src/commands/section/index.ts]: [P3] The last two examples are nearly identical. Consider updating these to match the `SKILL_CONTENT` examples which also demonstrate `--dry-run` and `--json` usage.

File: src/commands/section/reorder.ts [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    const [removed] = newOrder.splice(oldIndex, 1)
+    newOrder.splice(newIndex, 0, removed)
+
+    const items = newOrder.map((section, index) => ({
+        id: section.id,
+        sectionOrder: index + 1,
+    }))
+
+    if (options.dryRun) {
+        console.log(`Would reorder "${target.name}": position ${oldIndex} -> ${newIndex}`)
```
Comment (doistbot) [inline src/commands/section/reorder.ts]: [P3] For visual consistency with `project reorder`, consider using the unicode arrow `→` instead of `->` here (and similarly on line 162, as well as replacing `>` with `→` on line 147).

File: src/commands/section/reorder.ts [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+    after?: string
+    position?: number
+    json?: boolean
+    dryRun?: boolean
+}
+
+async function loadProjectSections(api: TodoistApi, projectId: string): Promise<Section[]> {
+    const { results } = await paginate(
+        (cursor, limit) => api.getSections({ projectId, cursor: cursor ?? undefined, limit }),
+        { limit: Number.MAX_SAFE_INTEGER },
```
Comment (doistbot) [inline src/commands/section/reorder.ts]: [P2] Using `paginate(..., { limit: Number.MAX_SAFE_INTEGER })` makes this path allocate the full `all` array and then clone it again inside `paginate()` via `slice(0, limit)` before returning. Since reorder always needs the complete project order, that doubles the memory/copy work for every run. Please switch this to a dedicated "load all pages" loop (or a helper that returns the accumulated array directly) instead of the huge-limit sentinel here.

File: src/commands/section/section.test.ts [hunk 6]
```diff
@@ -1,15 +1,22 @@
 import { Command } from 'commander'
-import { beforeEach, describe, expect, it, vi } from 'vitest'
+import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest'
 
 vi.mock('../../lib/api/core.js', () => ({
     getApi: vi.fn(),
 }))
 
+vi.mock('../../lib/api/sections-sync.js', () => ({
```
Comment (doistbot) [inline src/commands/section/section.test.ts]: [P2] The test relies on unnecessary module mocking of the internal `sections-sync.js` helper instead of mocking the SDK boundary. This creates a brittle coupling to the internal implementation rather than verifying the API payload. Please remove this mock, add `sync: vi.fn()` to `createMockApi()` in `src/test-support/mock-api.ts`, and assert against `mockApi.sync` using `expect.objectContaining` to match the SDK `createCommand` payload.

File: src/commands/section/section.test.ts [hunk 7]
```diff
@@ -264,6 +271,314 @@ describe('section update --json', () => {
...
+                'Review',
+                '--project',
+                'id:proj-work',
+                '--before',
+                'Review',
+            ]),
+        ).rejects.toHaveProperty('code', 'INVALID_OPTIONS')
+    })
+
+    it('--dry-run does not call reorderSections', async () => {
```
Comment (doistbot) [inline src/commands/section/section.test.ts]: [P2] This `--dry-run` test only verifies that `reorderSections` is skipped and that some log line is printed. Since the point of `--dry-run` is the preview itself, add assertions for the `New section order` output (or the listed ids/names) so a regression in the preview ordering can't still pass by logging only `Would reorder ...`.

File: src/commands/section/section.test.ts [hunk 8]
```diff
@@ -264,6 +271,314 @@ describe('section update --json', () => {
...
+            '--project',
+            'id:proj-work',
+            '--position',
+            '1',
+        ])
+
+        expect(mockReorderSections).not.toHaveBeenCalled()
+    })
+
+    it('--json outputs the new ordering', async () => {
```
Comment (doistbot) [inline src/commands/section/section.test.ts:592]: [P2] The suite exercises `--json` only for a real move, but `reorderSection` has a separate `newIndex === oldIndex && options.json` branch. Add a no-op JSON test here so `section reorder --json` is covered when nothing changes and can't regress to emitting the human `No change` message instead of the current ordering.

**Review by doistbot — COMMENTED** [review]
This PR introduces the `td section reorder` command, offering robust functionality to rearrange sections within a project using positional or relative moves. It is a great addition that significantly expands the CLI's project management capabilities. A few adjustments are noted regarding the reuse of existing entity resolution and validation helpers, optimizing pagination logic to reduce memory overhead, polishing output formatting for consistency, and strengthening the test suite with more robust SDK boundary mocks and coverage for edge cases.

<sub>[Share Feedback](https://github.com/Doist/doistbot-app/issues/new?template=review-feedback.yml&title=Review+feedback+for+Doist%2Ftodoist-cli&pr_url=https%3A%2F%2Fgithub.com%2FDoist%2Ftodoist-cli%2Fpull%2F349&datadog_url=https%3A%2F%2Fapp.datadoghq.com%2Flogs%3Fquery%3Dservice%253A%2528doistbot-app-sandbox%2520OR%2520doistbot-app%2529%2520%2540meta.deliveryId%253A74e933c0-5308-11f1-8a3d-c8b75ea4ceba%26agg_m%3Dcount%26agg_m_source%3Dbase%26agg_t%3Dcount%26clustering_pattern_field_path%3Dmessage%26cols%3Dhost%252Cservice%26fromUser%3Dtrue%26messageDisplay%3Dinline%26refresh_mode%3Dsliding%26storage%3Dhot%26stream_sort%3Ddesc%26viz%3Dstream%26live%3Dtrue) • [Review Logs](https://app.datadoghq.com/logs?query=service%3A%28doistbot-app-sandbox%20OR%20doistbot-app%29%20%40meta.deliveryId%3A74e933c0-5308-11f1-8a3d-c8b75ea4ceba&agg_m=count&agg_m_source=base&agg_t=count&clustering_pattern_field_path=message&cols=host%2Cservice&fromUser=true&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=stream&live=true)</sub>

File: src/commands/section/reorder.ts (hunk 2 above)
Comment (rmartins90) [reply src/commands/section/reorder.ts]: Extracted shared reorder placement validation and wired both project and section reorder through it.

File: src/commands/section/reorder.ts (hunk 1 above)
Comment (rmartins90) [reply src/commands/section/reorder.ts:37]: Updated to reuse the shared resolver and keep exact-match ambiguity handling there.

File: src/commands/section/reorder.ts (hunk 4 above)
Comment (rmartins90) [reply src/commands/section/reorder.ts]: Updated section reorder output to use the same `→` marker as project reorder.

File: src/commands/section/index.ts (hunk 3 above)
Comment (rmartins90) [reply src/commands/section/index.ts]: Updated the examples to cover `--dry-run` and `--json`.

File: src/commands/section/reorder.ts (hunk 5 above)
Comment (rmartins90) [reply src/commands/section/reorder.ts]: Switched this to an explicit all-pages loop.

File: src/commands/section/section.test.ts (hunk 8 above)
Comment (rmartins90) [reply src/commands/section/section.test.ts:592]: Added no-op `--json` coverage.

File: src/commands/section/section.test.ts (hunk 7 above)
Comment (rmartins90) [reply src/commands/section/section.test.ts]: Added explicit preview-order assertions.

File: src/commands/section/section.test.ts (hunk 6 above)
Comment (rmartins90) [reply src/commands/section/section.test.ts]: Updated the tests to assert the SDK sync payload instead of mocking the internal helper.

File: src/commands/section/reorder.ts [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+        sectionOrder: index + 1,
+    }))
+
+    if (options.dryRun) {
+        console.log(`Would reorder "${target.name}": position ${oldIndex} → ${newIndex}`)
+        console.log('New section order:')
+        for (let index = 0; index < newOrder.length; index++) {
+            const marker = newOrder[index].id === target.id ? '→' : ' '
+            console.log(`  ${marker} ${index}: ${newOrder[index].name} (id:${newOrder[index].id})`)
+        }
```
Comment (scottlovegrove) [inline src/commands/section/reorder.ts]: Why do we do this for the dry run, but not for the actual run? Shouldn't this match what we display on the actual run?

File: src/commands/section/section.test.ts [hunk 10]
```diff
@@ -264,6 +280,391 @@ describe('section update --json', () => {
...
+        await expect(
+            program.parseAsync([
+                'node',
+                'td',
+                'section',
+                'reorder',
+                'Review',
+                '--project',
+                'id:proj-work',
+            ]),
```
Comment (scottlovegrove) [inline src/commands/section/section.test.ts]: We could create a const here for the array as these are used in a lot of the tests and could act as the base args.

**Review by scottlovegrove — APPROVED** [review]

File: src/commands/section/section.test.ts (hunk 10 above)
Comment (rmartins90) [reply src/commands/section/section.test.ts]: Done, added shared base args for the reorder tests.

File: src/commands/section/reorder.ts (hunk 9 above)
Comment (rmartins90) [reply src/commands/section/reorder.ts]: Yeah, that makes sense. Updated the real run to print the new section order through the same helper as dry-run.
